integration/docker_engine_test.go

```diff
@@ -0,0 +1,73 @@
+// +build integration
+
+package integration
+
+import (
+	"context"
+	"io/ioutil"
+	"os"
+	"testing"
+
+	"github.com/aquasecurity/trivy/internal"
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/client"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestRun_WithDockerEngine(t *testing.T) {
+	testCases := []struct {
+		name               string
+		testfile           string
+		expectedOutputFile string
+		expectedError      error
+	}{
+		{
+			name:               "happy path, alpine:3.10",
+			expectedOutputFile: "testdata/alpine-310.json.golden",
+			testfile:           "testdata/fixtures/alpine-310.tar.gz",
+		},
+	}
+
+	for _, tc := range testCases {
+		ctx := context.Background()
+		testfile, _ := os.Open(tc.testfile)
+		cli, _ := client.NewClientWithOpts(client.FromEnv)
```

Comment:
I'm worried that the failing test will pass by accident due to ignoring an error. I'd like to add `require.NoError` just in case.

```suggestion
		cli, err := client.NewClientWithOpts(client.FromEnv)
		require.NoError(t, err, tc.name)
```

Reply:
It would panic on line 35 because cli will be nil, which is OK since that's a sign of a failing test. But sure I can add it in.